.github/workflows/validate-migration.yaml

```diff
@@ -5,49 +5,107 @@ on: [ push ]
+  VALGRIND_INSTALLATION_DIR: "/usr"
+  VALGRIND_SOURCE_DIR: "/tmp/valgrind-src"
+
   # Perforce credentials
   P4PORT: "perforce.sgdev.org:1666" # the address of the Perforce server to connect to
   P4USER: "admin" # the name of the Perforce user
   P4PASSWD: ${{ secrets.P4PASSWD }} # the ticket for the Perforce user
 
   DEPOT_NAME: "source/src-cli" # the name of the Perforce depot to validate the git migration against
 
+  # Enable sccache
+  SCCACHE_GHA_ENABLED: "true"
+
 jobs:
   validate-migration:
     strategy:
       fail-fast: false
       matrix:
-        sanitizer:
-          - addressSanitizer
-          - threadSanitizer
-          #          disabled until we figure out how to build with clang in github actions
-          #          - memorySanitizer
-          - leakSanitizer
-          - undefinedSanitizer
+        tool: # Note: If you add or change the name of tools here, you must also adjust the logic in the "Prepare environment for specific tool" step
+          - addressSanitizer # https://github.com/google/sanitizers/wiki/AddressSanitizer
+          - threadSanitizer # https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual
+          - undefinedSanitizer # https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
+          - valgrindMemcheck # https://valgrind.org/info/tools.html#memcheck
+          - valgrindHelgrind # https://valgrind.org/info/tools.html#helgrind
           - noSanitizer
-        include:
-          - sanitizer: addressSanitizer
-            CMAKE_CXX_FLAGS: "-fsanitize=address -g"
-          - sanitizer: threadSanitizer
-            CMAKE_CXX_FLAGS: "-fsanitize=thread -g"
-          #          disabled until we figure out how to build with clang in github actions
-          #          - sanitizer: memorySanitizer
-          #            CMAKE_CXX_FLAGS: "-fsanitize=memory -g"
-          - sanitizer: leakSanitizer
-            CMAKE_CXX_FLAGS: "-fsanitize=leak -g"
-          - sanitizer: undefinedSanitizer
-            CMAKE_CXX_FLAGS: "-fsanitize=undefined -g"
-          - sanitizer: noSanitizer
-            CMAKE_CXX_FLAGS: ""
 
     runs-on: ubuntu-latest
 
     steps:
+      - name: "Prepare environment for specific tool"
+        run: |
+          # Set environment variables for future steps based on the tool being used. We can't refer to the matrix variables in the env section
+          # of the job, so we have to do it here.
+          #
+          # See https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable 
+          # for more information on the method to set environment variables.
+          
+          bash <<EOF
+            set -euxo pipefail
+          
+            case ${{ matrix.tool }} in
+            
+            noSanitizer)
+              echo "USE_VALGRIND=false" >> $GITHUB_ENV
+              echo "CMAKE_CXX_FLAGS=" >> $GITHUB_ENV
```

Comment:
I was hestiant to do this since valgrind's documentation recommends using `-O0`, but it does run twice as fast under memcheck with `-O1`. I'll leave a comment in the workflow definition saying that if `memcheck` reports an issue but it's hard to follow the stack trace - then it's worth probably temporarily shifting the level back to `-O0`.